### PR TITLE
#174: Belos: Provide `Tpetra` version of BlockCG examples

### LINKED ISSUE #174: Belos: Provide `Tpetra` version of BlockCG examples

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #174

### CODE REVIEW

File: packages/belos/tpetra/example/BlockCG/PseudoBlockCGTpetraExFile.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+TEUCHOS_STANDARD_CATCH_STATEMENTS(verbose, std::cerr, success);
+
+return success ? EXIT_SUCCESS : EXIT_FAILURE;
+}
+
+int main(int argc, char *argv[]) {
+  // run with different ST
+  run<double>(argc,argv);
+  // run<float>(argc,argv); // FAILS
+}
```
Comment (tlamonthezie) [inline packages/belos/tpetra/example/BlockCG/PseudoBlockCGTpetraExFile.cpp]: Hello @cwschilly just a remark: the main function should return the result of the run call. Coul you please add a `return` statement to the `run` function call.

Comment (cwschilly) [reply packages/belos/tpetra/example/BlockCG/PseudoBlockCGTpetraExFile.cpp]: Fixed

**Comment by cwschilly** [comment]
Currently, this example does not converge

**Comment by tlamonthezie** [comment]
> Currently, this example does not converge

Hello @cwschilly yes it causes a timeout. I am looking at the problem

**Comment by tlamonthezie** [comment]
> > Currently, this example does not converge
> 
> Hello @cwschilly yes it causes a timeout. I am looking at the problem

After some tests, the example works if we use `Tpetra::Utils::readHBMatrix` instead of `Belos::Tpetra::HarwellBoeingReader` (based on TriUtils HB reader).
- Tests and example works with `orsirr1.hb` data file if read by the Belos::Tpetra::HarwellBoeingReader (#180 )
- Tests and examples works with `bcsstk14.hb` if using the Tpetra::Utils::readHBMatrix (similar problem encountered in #180)
So I don't know from where comes the problem. What is strange is that the pseudo example works correctly using the ` Belos::Tpetra::HarwellBoeingReader` to read `bcsstk14.hb`
@cwschilly could you please take a look at last commit in fact it is now like in the test named `Tpetra_BlockCG_hb_test` which looks like to be the same than the `BlockCG_Tpetra_File_Ex` example from this PR

**Comment by cwschilly** [comment]
@stmcgovern This PR is ready for review (pending CI completion)

**Review by stmcgovern — APPROVED** [review]

**Pull request merged**